Sources/FirebaseFirestore/DocumentReference+Swift.swift

```diff
@@ -41,15 +42,24 @@ extension DocumentReference {
 
     return snapshot.pointee
   }
-}
 
-extension DocumentReference: CustomDebugStringConvertible {
-  public var debugDescription: String {
-    String(self.ToString())
+  public func addSnapshotListener(_ listener: @escaping SnapshotListenerCallback) -> ListenerRegistration? {
+    addSnapshotListener(includeMetadataChanges: false, listener: listener)
+  }
+
+  public func addSnapshotListener(includeMetadataChanges: Bool, listener: @escaping SnapshotListenerCallback) -> ListenerRegistration? {
+    let boxed = Unmanaged.passRetained(listener as AnyObject)
+    let internalRegistration = swift_firebase.swift_cxx_shims.firebase.firestore.document_add_snapshot_listener(self, { snapshot, errorCode, errorMessage, pvListener in
```

Comment:
WDYT of `registration` rather than `internalRegistration`?  The wrapper is constructed at the very end.

Reply:
I dig it!